# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1357832837 18000
#      Thu Jan 10 10:47:17 2013 -0500
# Node ID d4671167972b8f1364b9b0c7c804378a879af168
# Parent  b91fc9348e2e723e2a605b16ff8aac08586ff52f
* etc/HACKING: Update with new gnulib info.

diff --git a/etc/HACKING b/etc/HACKING
--- a/etc/HACKING
+++ b/etc/HACKING
@@ -43,20 +43,20 @@ relevant information on downloading sour
 
   http://www.octave.org/download.html
 
 After cloning Octave, you will need to run the bootstrap script:
 
   $ ./bootstrap
 
 This script will examine the source tree and generate some Makefile
-fragments and then runs the bootstrap script.  The bootstrap script comes
-from gnulib, but is kept in the Octave source archive.  It should be
-updated from the gnulib sources as necssary.  The bootstrap script takes
-care of running the autotools and generating the configure script.
+fragments, then run autotools scripts to generate Makefile.in files from
+Makefile.am files and create the configure script.  The bootstrap script
+comes from gnulib, but is kept in the Octave source archive.  It should
+be updated from the gnulib sources as necessary.
 
 If you have a copy of gnulib in some directory apart from the Octave
 source tree, then pass the name of the directory containing gnulib-tool
 to the bootstrap script using the option: 
 
   --gnulib-srcdir=DIRNAME
 
 If you have downloaded gnulib directly, DIRNAME will be the directory where
@@ -68,20 +68,25 @@ By using an external gnulib directory, y
 tree among several projects.  Since 2011, the gnulib sources are a Mercurial
 subrepository, so they will be automatically updated to the
 corresponding Mercurial revision if you update the working directory to
 a past revision not too far in the past.
 
 Additional options besides --gnulib-srcdir can be passed to bootstrap and
 they will be forwarded without modification to the gnulib bootstrap script.
 
-Once the bootstrap and bootstrap_gnulib scripts complete successfully, you may
-run
+Once the bootstrap script completes successfully, you may configure and
+build Octave.  We recommend that you build Octave in a separate
+directory tree from the sources.  For example, if you have just finished
+running the bootstrap script in the top-level source directory, run the
+following commands to create a build tree, configure, and build Octave:
 
-  $ ./configure
+  $ mkdir .build
+  $ cd .build
+  $ ../configure
   $ make
   $ make check
 
 At this point, there should be no difference between your working tree
 and the currently visited hg revision:
 
   $ hg diff
 
@@ -104,23 +109,29 @@ and preparing patches.
 An overview of the directory layout of Octave's source files:
 
   doc           -- Texinfo documentation for Octave
 
 
   examples      -- some example files
 
 
-  gnulib        -- gnulib subrepo.  This is the actual gnulib source
-                   tree, checked out with git.
+  gnulib-hg     -- gnulib subrepo.  This is a clone of the gnulib source
+                   tree maintained by the Octave project.  The default
+                   branch is identical to the upstream gnulib sources.
+                   There is also an "octave-stable" branch that may
+                   contain changes as needed for the "stable" branch in
+                   the Octave archive.  We usually don't want to update
+                   gnulib sources when going from one stable point
+                   release to the next, but we occasionally need to
+                   include small updates.
 
 
   libgnu        -- gnulib sources that we use.  The files here are
                    copied here from the gnulib directory by the
-                   build-aux/bootstrap_gnulib script that is run by the
                    bootstrap script.
 
 
   liboctave     -- C++ interfaces to the numerical libraries, Fortran
                    numerical libraries, various OS facilities, and utility
                    functions
 
     array       the base Array, NDArray, Matrix, and Sparse classes
@@ -289,17 +300,17 @@ An overview of the directory layout of O
     fntests.m        script to run function tests embedded in C++ and .m
                      files
 
 ----
 John W. Eaton
 jwe@octave.org
 
 
-Last updated: Sat, 18 Aug 2012 18:51:25 EDT
+Last updated: Thu, 10 Jan 2013 10:46:41 EST
 
 
 ################################################################################
 
 Copyright (C) 2009-2012 John W. Eaton
 
 This file is part of Octave.
 
